tests/suite/scale_test.go

```diff
@@ -728,66 +796,386 @@ var _ = Describe("Scale test", Ordered, Label("nfr", "scale"), func() {
+		if *plusEnabled {
+			link = fmt.Sprintf("\n\n![%[1]v-plus.png](%[1]v-plus.png)\n", fmt.Sprintf("%s-%s", testFileNamePrefix, res.scheme))
+		}
+
+		_, err = fmt.Fprintf(outFile, "```%s", link)
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	BeforeAll(func() {
+		// Scale tests need a dedicated NGF instance
+		// Because they analyze the logs of NGF and NGINX, and they don't want to analyze the logs of other tests.
+		teardown(releaseName)
+
+		var err error
+		resultsDir, err = framework.CreateResultsDir("zero-downtime-scale", version)
+		Expect(err).ToNot(HaveOccurred())
+
+		filename := filepath.Join(resultsDir, framework.CreateResultsFilename("md", version, *plusEnabled))
+		outFile, err = framework.CreateResultsFile(filename)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(framework.WriteSystemInfoToFile(outFile, clusterInfo, *plusEnabled)).To(Succeed())
+	})
+
+	AfterAll(func() {
+		_, err := fmt.Fprintln(outFile)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(outFile.Close()).To(Succeed())
+
+		// restoring NGF shared among tests in the suite
+		cfg := getDefaultSetupCfg()
+		cfg.nfr = true
+		setup(cfg)
+	})
+
+	BeforeEach(func() {
+		ns = core.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "zero-downtime-scale",
+			},
+		}
+
+		metricsCh = make(chan *metricsResults, 2)
+	})
+
+	tests := []struct {
+		name        string
+		valuesFile  string
+		numReplicas int
+	}{
+		{
+			name:        "One NGF Pod runs per node",
+			valuesFile:  "manifests/scale/zero-downtime/values-affinity.yaml",
+			numReplicas: 12, // equals number of nodes
+		},
+		{
+			name:        "Multiple NGF Pods run per node",
+			valuesFile:  "manifests/scale/zero-downtime/values.yaml",
+			numReplicas: 24, // twice the number of nodes
+		},
 	}
 
-	return tmpl.Execute(dest, results)
-}
+	for _, test := range tests {
+		When(test.name, func() {
+			BeforeAll(func() {
+				cfg := getDefaultSetupCfg()
+				cfg.nfr = true
+				setup(cfg, "--values", test.valuesFile)
+
+				deploy, err := resourceManager.GetNGFDeployment(ngfNamespace, releaseName)
+				Expect(err).ToNot(HaveOccurred())
+				ngfDeploymentName = deploy.GetName()
+
+				Expect(resourceManager.Apply([]client.Object{&ns})).To(Succeed())
+				Expect(resourceManager.ApplyFromFiles(files, ns.Name)).To(Succeed())
+				Expect(resourceManager.WaitForAppsToBeReady(ns.Name)).To(Succeed())
+
+				_, err = fmt.Fprintf(outFile, "\n## %s Test Results\n", test.name)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			AfterAll(func() {
+				teardown(releaseName)
+				Expect(resourceManager.DeleteNamespace(ns.Name)).To(Succeed())
+			})
+
+			It("scales up gradually without downtime", func() {
+				_, err := fmt.Fprint(outFile, "\n### Scale Up Gradually\n")
+				Expect(err).ToNot(HaveOccurred())
+
+				testFileNamePrefix := formatTestFileNamePrefix("gradual-scale-up", test.valuesFile)
+
+				for _, test := range trafficConfigs {
+					wg.Add(1)
```

Comment:
Does the wait group still need to be a global var? Probably better to create it in each function